upgraderesponder/service.go

```diff
@@ -77,8 +77,16 @@ type Version struct {
 }
 
 type CheckUpgradeRequest struct {
-	AppVersion string            `json:"appVersion"`
-	ExtraInfo  map[string]string `json:"extraInfo"`
+	AppVersion       string            `json:"appVersion"`
+	HarvesterVersion string            `json:"harvesterVersion"`
+	ExtraInfo        map[string]string `json:"extraInfo"`
+}
+
+func (r *CheckUpgradeRequest) GetVersion() string {
+	if r.AppVersion != "" || r.HarvesterVersion == "" {
+		return r.AppVersion
+	}
+	return r.HarvesterVersion
```

Comment:
Is there different beetween the above and this logic?
```suggestion
	if r.AppVersion != "" {
		return r.AppVersion
	}
	return r.HarvesterVersion
```

Reply:
I change the logic to the following. Do you think it's more clear?  Thank you.

```go
// GetVersion returns version from AppVersion and HarvesterVersion.
// Only if AppVersion is empty and HarvesterVersion is not, we return HarvesterVersion.
// We will remove this function after all customers' Harvester version update to v1.1.0.
func (r *CheckUpgradeRequest) GetVersion() string {
	if r.AppVersion == "" && r.HarvesterVersion != "" {
		return r.HarvesterVersion
	}
	return r.AppVersion
}
```

Reply:
I the future, Longhorn also need to migrate to this newer format. So, how di you think about 

```go
// GetVersion returns version from AppVersion and HarvesterVersion.
// Only if AppVersion is empty and HarvesterVersion is not, we return HarvesterVersion.
// We will remove this function after all customers' Harvester version update to v1.1.0.
func (r *CheckUpgradeRequest) GetVersion() string {
	if r.AppVersion != "" {
		return r.AppVersion
	}
        // for future Longhorn migration
        //  if r.LonghornVersion != "" {
	//	 return r.LonghornVersion
	// }
        if r.HarvesterVersion != "" {
		return r.HarvesterVersion
	}
	return ""

}
```

Reply:
Yes, this one looks good to me. I update it. Thank you.